### ISSUE TITLE
Some problems when i train VOC2007 database

### ISSUE THREAD

**Issue by EvanYyf** [ShaoqingRen/faster_rcnn#193]
opts:
      cache_name: 'fast_rcnn_VOC2007_ZF'
            conf: [1×1 struct]
    ignore_cache: 0
            imdb: [1×1 struct]
    net_def_file: 'F:\yyf\faster_rcnn-master\faster_rcnn-master\models\fast_rcnn_prototxts\ZF\test.prototxt'
        net_file: 'F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\final'
           roidb: [1×1 struct]
          suffix: ''

conf:
       batch_size: 128
      bbox_thresh: 0.5000
     bg_thresh_hi: 0.5000
     bg_thresh_lo: 0.1000
      fg_fraction: 0.2500
        fg_thresh: 0.5000
      image_means: [224×224×3 single]
    ims_per_batch: 2
         max_size: 1000
         rng_seed: 6
           scales: 600
      test_binary: 0
    test_max_size: 1000
         test_nms: 0.3000
      test_scales: 600
      use_flipped: 1
          use_gpu: 1

faster_rcnn-master: test (voc_2007_test) 1/4952  time: 0.400s
.....The middle part is omitted.......
faster_rcnn-master: test (voc_2007_test) 1000/4952  time: 0.039s
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf

faster_rcnn-master: test (voc_2007_test) 1001/4952  time: 0.037s
.....The middle part is omitted.......
faster_rcnn-master: test (voc_2007_test) 3000/4952  time: 0.036s
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf

faster_rcnn-master: test (voc_2007_test) 3001/4952  time: 0.031s
.....The middle part is omitted.......
faster_rcnn-master: test (voc_2007_test) 4000/4952  time: 0.034s
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf

faster_rcnn-master: test (voc_2007_test) 4001/4952  time: 0.036s
.....The middle part is omitted.......
faster_rcnn-master: test (voc_2007_test) 4952/4952  time: 0.037s
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf
  -Inf

test all images in 182.725756 seconds.
Cleared 0 solvers and 1 stand-alone nets
aeroplane: pr: load: 639/4952
aeroplane: pr: load: 1275/4952
aeroplane: pr: load: 1870/4952
aeroplane: pr: load: 2495/4952
aeroplane: pr: load: 3092/4952
aeroplane: pr: load: 3691/4952
aeroplane: pr: load: 4279/4952
aeroplane: pr: load: 4892/4952
!!! aeroplane : 0.0000 0.0000
bicycle: pr: load: 638/4952
bicycle: pr: load: 1253/4952
bicycle: pr: load: 1850/4952
bicycle: pr: load: 2470/4952
bicycle: pr: load: 3074/4952
bicycle: pr: load: 3673/4952
bicycle: pr: load: 4262/4952
bicycle: pr: load: 4871/4952
!!! bicycle : 0.0000 0.0000
bird: pr: load: 640/4952
bird: pr: load: 1279/4952
bird: pr: load: 1877/4952
bird: pr: load: 2476/4952
bird: pr: load: 2991/4952
bird: pr: load: 3570/4952
bird: pr: load: 4168/4952
bird: pr: load: 4774/4952
!!! bird : 0.0000 0.0000
boat: pr: load: 639/4952
boat: pr: load: 1270/4952
boat: pr: load: 1868/4952
boat: pr: load: 2487/4952
boat: pr: load: 3081/4952
boat: pr: load: 3654/4952
boat: pr: load: 4237/4952
boat: pr: load: 4839/4952
!!! boat : 0.0000 0.0000
bottle: pr: load: 640/4952
bottle: pr: load: 1278/4952
bottle: pr: load: 1874/4952
bottle: pr: load: 2499/4952
bottle: pr: load: 3092/4952
bottle: pr: load: 3691/4952
bottle: pr: load: 4278/4952
bottle: pr: load: 4862/4952
!!! bottle : 0.0000 0.0000
bus: pr: load: 640/4952
bus: pr: load: 1276/4952
bus: pr: load: 1872/4952
bus: pr: load: 2493/4952
bus: pr: load: 3089/4952
bus: pr: load: 3679/4952
bus: pr: load: 4270/4952
bus: pr: load: 4874/4952
!!! bus : 0.0000 0.0000
car: pr: load: 640/4952
car: pr: load: 1277/4952
car: pr: load: 1872/4952
car: pr: load: 2493/4952
car: pr: load: 3091/4952
car: pr: load: 3677/4952
car: pr: load: 4269/4952
car: pr: load: 4874/4952
!!! car : 0.0000 0.0000
cat: pr: load: 640/4952
cat: pr: load: 1279/4952
cat: pr: load: 1876/4952
cat: pr: load: 2497/4952
cat: pr: load: 3092/4952
cat: pr: load: 3688/4952
cat: pr: load: 4274/4952
cat: pr: load: 4877/4952
!!! cat : 0.0000 0.0000
chair: pr: load: 640/4952
chair: pr: load: 1276/4952
chair: pr: load: 1874/4952
chair: pr: load: 2481/4952
chair: pr: load: 3077/4952
chair: pr: load: 3664/4952
chair: pr: load: 4249/4952
chair: pr: load: 4855/4952
!!! chair : 0.0000 0.0000
cow: pr: load: 640/4952
cow: pr: load: 1279/4952
cow: pr: load: 1873/4952
cow: pr: load: 2496/4952
cow: pr: load: 3091/4952
cow: pr: load: 3682/4952
cow: pr: load: 4271/4952
cow: pr: load: 4874/4952
!!! cow : 0.0000 0.0000
diningtable: pr: load: 637/4952
diningtable: pr: load: 1267/4952
diningtable: pr: load: 1865/4952
diningtable: pr: load: 2484/4952
diningtable: pr: load: 3081/4952
diningtable: pr: load: 3674/4952
diningtable: pr: load: 4263/4952
diningtable: pr: load: 4871/4952
!!! diningtable : 0.0000 0.0000
dog: pr: load: 638/4952
dog: pr: load: 1270/4952
dog: pr: load: 1868/4952
dog: pr: load: 2487/4952
dog: pr: load: 3085/4952
dog: pr: load: 3679/4952
dog: pr: load: 4270/4952
dog: pr: load: 4874/4952
!!! dog : 0.0000 0.0000
horse: pr: load: 640/4952
horse: pr: load: 1279/4952
horse: pr: load: 1878/4952
horse: pr: load: 2504/4952
horse: pr: load: 3096/4952
horse: pr: load: 3692/4952
horse: pr: load: 4279/4952
horse: pr: load: 4879/4952
!!! horse : 0.0000 0.0000
motorbike: pr: load: 640/4952
motorbike: pr: load: 1281/4952
motorbike: pr: load: 1880/4952
motorbike: pr: load: 2505/4952
motorbike: pr: load: 3098/4952
motorbike: pr: load: 3693/4952
motorbike: pr: load: 4279/4952
motorbike: pr: load: 4889/4952
!!! motorbike : 0.0000 0.0000
person: pr: load: 640/4952
person: pr: load: 1279/4952
person: pr: load: 1879/4952
person: pr: load: 2505/4952
person: pr: load: 3099/4952
person: pr: load: 3695/4952
person: pr: load: 4283/4952
person: pr: load: 4896/4952
!!! person : 0.0000 0.0000
pottedplant: pr: load: 640/4952
pottedplant: pr: load: 1281/4952
pottedplant: pr: load: 1883/4952
pottedplant: pr: load: 2506/4952
pottedplant: pr: load: 3103/4952
pottedplant: pr: load: 3699/4952
pottedplant: pr: load: 4284/4952
pottedplant: pr: load: 4898/4952
!!! pottedplant : 0.0000 0.0000
sheep: pr: load: 639/4952
sheep: pr: load: 1267/4952
sheep: pr: load: 1864/4952
sheep: pr: load: 2484/4952
sheep: pr: load: 3081/4952
sheep: pr: load: 3675/4952
sheep: pr: load: 4265/4952
sheep: pr: load: 4865/4952
!!! sheep : 0.0000 0.0000
sofa: pr: load: 640/4952
sofa: pr: load: 1280/4952
sofa: pr: load: 1878/4952
sofa: pr: load: 2505/4952
sofa: pr: load: 3092/4952
sofa: pr: load: 3691/4952
sofa: pr: load: 4268/4952
sofa: pr: load: 4870/4952
!!! sofa : 0.0000 0.0000
train: pr: load: 639/4952
train: pr: load: 1271/4952
train: pr: load: 1865/4952
train: pr: load: 2484/4952
train: pr: load: 3082/4952
train: pr: load: 3676/4952
train: pr: load: 4269/4952
train: pr: load: 4874/4952
!!! train : 0.0000 0.0000
tvmonitor: pr: load: 640/4952
tvmonitor: pr: load: 1281/4952
tvmonitor: pr: load: 1883/4952
tvmonitor: pr: load: 2506/4952
tvmonitor: pr: load: 3102/4952
tvmonitor: pr: load: 3699/4952
tvmonitor: pr: load: 4286/4952
tvmonitor: pr: load: 4899/4952
!!! tvmonitor : 0.0000 0.0000

~~~~~~~~~~~~~~~~~~~~
Results:
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0
     0

     0

~~~~~~~~~~~~~~~~~~~~

It seems that nothing had been computed ,how can i solve that?

**Comment by EvanYyf** [comment]
When the program goes to here,it starts to wrong.




Preparing training data...Done.
Preparing validation data...Done.

------------------------- Iteration 2000 -------------------------
Training : error 0.698, loss (cls 2.66, reg NaN)
Testing  : error 0.692, loss (cls 2.59, reg NaN)

------------------------- Iteration 4000 -------------------------
Training : error 0.704, loss (cls 2.6, reg NaN)
Testing  : error 0.692, loss (cls 2.59, reg NaN)

------------------------- Iteration 6000 -------------------------
Training : error 0.686, loss (cls 2.57, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 8000 -------------------------
Training : error 0.701, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 10000 -------------------------
Training : error 0.688, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.59, reg NaN)
Saved as F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\iter_10000

------------------------- Iteration 12000 -------------------------
Training : error 0.702, loss (cls 2.6, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 14000 -------------------------
Training : error 0.692, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.59, reg NaN)

------------------------- Iteration 16000 -------------------------
Training : error 0.686, loss (cls 2.56, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 18000 -------------------------
Training : error 0.696, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 20000 -------------------------
Training : error 0.692, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)
Saved as F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\iter_20000

------------------------- Iteration 22000 -------------------------
Training : error 0.69, loss (cls 2.57, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 24000 -------------------------
Training : error 0.696, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 26000 -------------------------
Training : error 0.696, loss (cls 2.58, reg NaN)
Testing  : error 0.692, loss (cls 2.59, reg NaN)

------------------------- Iteration 28000 -------------------------
Training : error 0.702, loss (cls 2.6, reg NaN)
Testing  : error 0.692, loss (cls 2.58, reg NaN)

------------------------- Iteration 30000 -------------------------
Training : error 0.689, loss (cls 2.56, reg NaN)
Testing  : error 0.682, loss (cls 2.55, reg NaN)
Saved as F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\iter_30000

------------------------- Iteration 32000 -------------------------
Training : error 0.677, loss (cls 2.52, reg NaN)
Testing  : error 0.679, loss (cls 2.5, reg NaN)

------------------------- Iteration 34000 -------------------------
Training : error 0.587, loss (cls 2.07, reg NaN)
Testing  : error 0.389, loss (cls 1.24, reg NaN)

------------------------- Iteration 36000 -------------------------
Training : error 0.325, loss (cls 1.05, reg NaN)
Testing  : error 0.266, loss (cls 0.867, reg NaN)

------------------------- Iteration 38000 -------------------------
Training : error 0.234, loss (cls 0.743, reg NaN)
Testing  : error 0.228, loss (cls 0.748, reg NaN)
Saved as F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\iter_40000
Saved as F:\yyf\faster_rcnn-master\faster_rcnn-master\output\fast_rcnn_cachedir\fast_rcnn_VOC2007_ZF\voc_2007_trainval\final
